### ISSUE TITLE
Webhook Connector: Support verification challenges

### ISSUE THREAD

**Issue by chillleader** [camunda/connectors#516]
## What do we need to do?

Support verification challenges in HTTP Webhook Connector.

The verification challenge aims to check whether the user has control over the webhook handler code by sending a request with a certain parameter. Webhook handler must reply with the same parameter value or perform some operation on it.

[Learn more about verification challenge](https://webhooks.fyi/security/one-time-verification-challenge)

## Breakdown

⚠️ Verification endpoints usually use GET requests - #515  is a prerequisite and a blocker for this issue.

- [x] Analyse how one-time verification is handled in real-life integrations and define the implementation scope accordingly
  - Proposed more or less universal solution: require the user to provide a FEEL expression that transforms any part of incoming request. The result of this expression is returned in response body.
  - Presumably, verification challenges almost always use GET requests and imply that regular webhook requests use POST. Therefore, we can assume that verification challenges can only be configured when the overall webhook method is POST.
- [x] Develop webhook element template changes to support verification challenge configuration, which may look something like this (depending on the implementation scope):<br><img width="384" alt="Screenshot 2023-03-30 at 12 55 02" src="https://user-images.githubusercontent.com/38818382/228814724-0633d455-065a-40e5-833d-b4b2399995c9.png">
- [x] On top of the changes implemented in #386, support the one-time verification flow in connector runtime:
  - [x] When verification challenge is enabled for a specific webhook, an additional GET webhook should be activated alongside the main POST webhook. This state should be stored in [`WebhookConnectorRegistry`](https://github.com/camunda-community-hub/spring-zeebe/blob/main/connector-runtime/src/main/java/io/camunda/connector/runtime/inbound/webhook/WebhookConnectorRegistry.java).
  - [x] Verification challenge endpoint should always reply with a result of a FEEL expression evaluation. FEEL expression should be configurable in an element template.
  - [x] When webhook is deactivated (e.g. new process definition is deployed without webhook connector), verification endpoint should be deactivated as well.

## Implementation

- [x] SDK change https://github.com/camunda/connector-sdk/pull/531
- [x] Bundle change https://github.com/camunda/connectors-bundle/pull/925
- [x] Update docs

## Why should we do it?

In many API integrations, like Slack or Meta/Facebook, it is required to complete the verification challenge. Lack of its support is a major limitation for our ability to support various real-world systems.

### Additional info

This issue was transferred from [Spring Zeebe](https://github.com/camunda-community-hub/spring-zeebe/issues/379) repo.

**Comment by sbuettner** [comment]
As discussed @chillleader will look at 2-3 implementations and we will discuss this in the architecture session.

**Comment by chillleader** [comment]
Here are a few examples of how verification challenges work on different platforms.

### WhatsApp Business Platform

**TL;DR** Verification challenge comes as a query parameter and needs to be echoed with body. There is also another query param that allows webhook consumer to prove the authenticity of the provider.

<details>
  <summary>Details for WhatsApp</summary>

https://developers.facebook.com/docs/graph-api/webhooks/getting-started#verification-requests

This is what verification request looks like:
```
GET https://www.your-clever-domain-name.com/webhooks?
  hub.mode=subscribe&
  hub.challenge=1158201444&
  hub.verify_token=meatyhamhock
```

`hub.challenge` needs to be echoed
`hub.verify_token` is set by developer in the admin console and can be ignored or checked. It verifies that request comes from WhatsApp.
  
</details>

### Zoom

**TL;DR** We receive a token in a POST request body and need to hash it using a shared secret and return in the response body.

<details>
  <summary>Details for Zoom</summary>

https://developers.zoom.us/docs/api/rest/webhook-reference/#validate-your-webhook-endpoint

This is received in body:

```
{
  "payload": {
    "plainToken": "qgg8vlvZRS6UYooatFL8Aw"
  },
  "event_ts": 1654503849680,
  "event": "endpoint.url_validation"
}
```

This is expected in response:

```
{
  "plainToken": "qgg8vlvZRS6UYooatFL8Aw",
  "encryptedToken": "23a89b634c017e5364a1c8d9c8ea909b60dd5599e2bb04bb1558d9c3a121faa5"
}
```

(HMAC in base64, shared secret)

</details>

### Okta

**TL;DR** Simple echo challenge in request headers

<details>
  <summary>Details for Okta</summary>

https://developer.okta.com/docs/concepts/event-hooks/#one-time-verification-request

Response format:
```{ "verification" : "value_from_header" }```

</details>


Given such a diversity, I think the common ground here might be to implement a simple support of a FEEL expression that extracts verification challenge from the request that will then be echoed.

⚠️ Note: currently query parameters are not part of FEEL context for any expression we already support. We will need to change the response model accordingly, so query params are also available for result expressions.

More difficult scenarios such as Zoom can always be implemented as custom connectors, where challenge is handled by Java code.
